pkg/bootstrap/bootstrap_config.go

```diff
@@ -292,3 +297,34 @@ func WriteBootstrap(config *meshconfig.ProxyConfig, node string, epoch int, pilo
 	err = t.Execute(fout, opts)
 	return fname, err
 }
+
+func getNodeIPs() string {
+	ipAddresses := ""
+
+	ifaces, _ := net.Interfaces()
+
+	for _, iface := range ifaces {
+		if iface.Flags&net.FlagUp == 0 {
+			continue // interface down
+		}
+		if iface.Flags&net.FlagLoopback != 0 {
+			continue // loopback interface
+		}
+		addrs, _ := iface.Addrs()
+
+		for _, addr := range addrs {
+			var ip net.IP
+			switch v := addr.(type) {
+			case *net.IPNet:
+				ip = v.IP
+			case *net.IPAddr:
+				ip = v.IP
+			}
+			if ip == nil || ip.IsLoopback() {
+				continue
+			}
+			ipAddresses = ipAddresses + ip.String() + ";"
+		}
+	}
+	return ipAddresses
+}
```

Comment:
there is a similar routine in pilot/pkg/proxy/net.go. Might be possible to merge the two?

Reply:
I looked into the codes again, these two routines should be merged.